node-wasm/src/utils.rs

```diff
@@ -186,29 +194,50 @@ pub(crate) async fn request_storage_persistence() -> Result<(), Error> {
 }
 
 const CHROME_USER_AGENT_DETECTION_STR: &str = "Chrome/";
+const FIREFOX_USER_AGENT_DETECTION_STR: &str = "Firefox/";
+const SAFARI_USER_AGENT_DETECTION_STR: &str = "Safari/";
```

Comment:
Are you talking about the version string that follows? Because I want to avoid regex.

Reply:
I was taking mostly about the tables with rules for browser detection (which strings it shoud/shouldn't contain). 

Also for websocket, we might want to do detection per-engine and match on webkit? Especially since once iOS browsers move off of webkit, they'll probably update user-agent to match and we'll pick that up automatically:
<img width="749" alt="Screenshot 2024-07-30 at 10 03 58 AM" src="https://github.com/user-attachments/assets/da09f268-7359-42ac-8a11-35a353affd4c">

Reply:
Not insisting on changing it here, but it'd be nice to have browser detection rules coming from some authoritative source, rather than some ad-hoc set of rules we've come up with.